packages/react-ui-components/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -1,53 +1,42 @@
-import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import Editor, { useMonaco } from '@monaco-editor/react';
-import { DEFAULT_EDITOR_LANGUAGE, DEFAULT_EDITOR_THEME, DEFAULT_PADDING_TOP, FONT_NOT_FOUND_ERROR } from './config';
+import { DEFAULT_EDITOR_LANGUAGE, DEFAULT_EDITOR_THEME, DEFAULT_PADDING_TOP, KELVIN_CODE_EDITOR_THEME } from './config';
 import { ECodeEditorTheme, ICodeEditorProps, OnEditorChangeCallback } from './types';
 import { KvLoader } from '../stencil-generated';
 import { getEditorOptions } from './utils';
+import { useFontsApi } from '../../utils/useFont/useFontsApi';
 
 export const KvCodeEditor = ({
 	code,
 	readOnly = false,
 	loadingComponent = <KvLoader isLoading />,
```

Comment:
This component will be recreated on every mount which is not ideal.
Can we create a "CodeEditorLoader" component and pass the reference to the component instead of react node?
Something like this:
```typescript
export const KvCodeEditor = ({
    ...
    LoadingComponent = CodeEditorLoader,
	...
}: ICodeEditorProps) => {
	...
	if (!checkFont()) {
		return <LoaderComponent />;
	}
    ...
};
```
Also, component props are usually capitalised to distinguish them from other props types.